install_config/install/docker_registry.adoc

```diff
@@ -790,6 +796,145 @@ It is recommended that configuration files be maintained in a source control
 repository.
 ====
 
+[[registry-configuration-reference]]
+=== Registry Configuration Reference
+
+There are many configuration options available in the upstream
+link:https://github.com/docker/distribution[docker distribution]
+library. Not all link:https://docs.docker.com/registry/configuration/[configuration options]
+are supported or enabled. Use this section as a reference.
+
+[TIP]
+Every configuration option in this file may also be overridden using environment
+variables. See upstream documentation on how to
+link:https://docs.docker.com/registry/configuration/#override-specific-configuration-options[override specific configuration options].
+
+==== log
+
+link:https://docs.docker.com/registry/configuration/#log[Upstream options] supported.
+Example:
+
+====
+----
+log:
+  level: debug
+  formatter: text
+  fields:
+    service: registry
+    environment: staging
+----
+====
+
+==== hooks
+
+Mail hooks are not supported.
+
+==== storage
+
+The following link:https://docs.docker.com/registry/configuration/#storage[storage drivers]
+are supported:
+
+- link:https://docs.docker.com/registry/storage-drivers/filesystem/[filesystem]
+- link:https://docs.docker.com/registry/storage-drivers/azure/[azure]
+- link:https://docs.docker.com/registry/storage-drivers/s3/[s3] See
+link:#middleware[Middleware] section below for CloudFront configuration
+- link:https://docs.docker.com/registry/storage-drivers/swift/[swift]
+
+Global registry storage configuration options:
+
+- link:https://docs.docker.com/registry/configuration/#maintenance[maintenance] unconfirmed
+- link:https://docs.docker.com/registry/configuration/#upload-purging[upload purging] unconfirmed
+- link:https://docs.docker.com/registry/configuration/#read-only-mode[read-only mode] unconfirmed
+- link:https://docs.docker.com/registry/configuration/#delete[delete] supported
+- link:https://docs.docker.com/registry/configuration/#cache[cache] supported
+- link:https://docs.docker.com/registry/configuration/#redirect[redirect] unconfirmed
+
+==== auth
+
+Auth options should not be altered. The **openshift** extension is the only supported
+option.
+
+----
+auth:
+  openshift:
+    realm: openshift
+----
+
+==== middleware
+
+The **repository** middleware extension should not be altered.
```

Comment:
only the options section should be altered.  For instance if you want to disable pullthrough for some reason